### ISSUE TITLE
Sometimes icons don't show up in the add-on UI

### ISSUE THREAD

**Issue by johngruen** [mozilla/testpilot#702]
@meandavejustice 

![screen shot 2016-05-03 at 2 07 55 pm](https://cloud.githubusercontent.com/assets/3323249/15021156/3b81f77c-11f3-11e6-8720-f9686ec7721e.png)

**Comment by meandavejustice** [comment]
@johngruen Can you check how you are reproducing this? Is this only on local development that you are seeing it?

**Comment by lmorchard** [comment]
I have a hunch that this might be caused by the pre-signed S3 URLs we've been using (and fixed in #836). IIRC, those pre-signed URLs expire after about 10 min. So, If your add-on loaded up the list of experiments from the API, and it took you a little bit to try looking at the list, those image requests could end up being "access denied" messages from AWS and so never appear.

TL;DR: It's possible this issue could go away after our next deployment.

**Comment by johngruen** [comment]
@meandavejustice @lmorchard definitely on prod. Willing to believe @lmorchard's diagnosis

**Comment by johngruen** [comment]
I also have not been able to repro this recently, (since launch day really)...maybe close it?

**Comment by meandavejustice** [comment]
We should wait until the pre-signed S3 fix is on prod. Is that the case yet? @lmorchard

**Comment by pdehaan** [comment]
I still see this occasionally on production, but I can't find reliable repro steps since it usually resolves itself after a refresh or reopening of the doorhanger.

**Comment by meandavejustice** [comment]
@pdehaan Have you seen this recently, I still haven't been able to reproduce

**Comment by pdehaan** [comment]
I've seen it a few times in the last couple days, but I'm always switching between dev+prod, so I'm not certain on what envs I was on at the time.

**Comment by meandavejustice** [comment]
I remember @6a68 and @chuckharmston talking about xul caching issues before, could that be a possibility here when switching envs?

**Comment by meandavejustice** [comment]
@ckprice I'm having trouble reproducing this bug. I don't think it will be fixed in tomorrows milestone, would you like to back it out?

**Comment by pdehaan** [comment]
I'm not sure of the _why_ this is a happenin', but one potential solution may be to see if we can just inline the SVGs into that panel as data-uris, or whatever.

Next time i see it, I'll leave the panel open for a couple minutes and see if it's potentially just an extreme latency issue.

**Comment by ckprice** [comment]
I'm going to remove the milestone here until we have solid STR.

**Comment by pdehaan** [comment]
STR is to open the Test Pilot door hanger. 5-10% of the time it doesn't show icons.
Closing the door hanger and reopening seems to fix it, but it's suboptimal workaround.

**Comment by johngruen** [comment]
@pdehaan you still seeing this?

**Comment by pdehaan** [comment]
I don't remember seeing it this week on dev or prod.

**Comment by ghost** [comment]
:expressionless: :no_mouth:
